frappe/model/document.py

```diff
@@ -647,6 +647,7 @@ def set_new_name(self, force=False, set_name=None, set_child_names=True):
 		"""Calls `frappe.naming.set_new_name` for parent and child docs."""
 
 		if (frappe.flags.api_name_set or self.flags.name_set) and not force:
+			frappe.flags.api_name_set = False
```

Comment:
@akhilnarang code smell. This should not be a global flag. Not possible to use `doc.flags`?